fmriprep/cli/run.py

```diff
@@ -26,7 +26,15 @@ def main():
     # CRITICAL Save the config to a file. This is necessary because the execution graph
     # is built as a separate process to keep the memory footprint low. The most
     # straightforward way to communicate with the child process is via the filesystem.
-    config_file = config.execution.work_dir / ".fmriprep.toml"
+    config_file = config.execution.work_dir / ".".join(
+        (
+            ".fmriprep",
+            config.execution.participant_label[0]
+            if config.execution.participant_label
+            else "",
+            "toml",
+        )
+    )
```

Comment:
I agree with @effigies this is prone to leave use-cases uncovered. E.g., running several runs on the same participant but different ``--bids-filters``. A solid way of making sure of this is using a temporary directory.

```suggestion
    config_file = Path(mkdtemp()) / ".fmriprep.toml"
```
(adding `from tempfile import mkdtemp` somewhere around lines 10-13).
I think at some point in the end, we copy the config file to a folder with the uuid under derivatives (like crashfiles). We should copy this temporal file and delete the folder then.

Reply:
I can't seem to find it (where we copy the config to the derivatives folder).

Anyways, it could be nice to store it in `config.execution.output_dir / "fmriprep" / "logs" / f"config-{config.execution.run_uuid}.toml"` or similar.

You could set this path at the very beginning, although there could be some implications related to #2107 and the reutilization of uuids. Other than that, since this is not I/O bound, writing to a temp folder will not buy us much and directly using a filename on the output folder (or work directory, either way) tagged with the uuid might be more effective.

Reply:
Sure, we can generate a run specific config file. I designed this with #2107 in mind, but I'm fine with relying on just the config in the output directory.

As for the output config path, I think I prefer the current implementation better (putting all the config, crashes, etc. in the same directory). We may however want to put the most recent config at:

`out / "fmriprep" / subject / "log" / "latest-fmriprep.toml"`
for ease of viewing

Reply:
> Sure, we can generate a run specific config file

I think this is a requirement, not a possibility here. Then, #2107 should work off of this bugfix.



> As for the output config path, I think I prefer the current implementation better (putting all the config, crashes, etc. in the same directory)

I'm fine with this. I just checked and we do write out a file `out / "fmriprep" / subject / "log" / uuid / "fmriprep.toml"`. So maybe it's just a matter of working only with this file.



> We may however want to put the most recent config at:
> 
> `out / "fmriprep" / subject / "log" / "latest-fmriprep.toml"`
> for ease of viewing

uuids have the date and time in the beginning. It's just a matter of picking the latest. From the user's perspective, this might be easier. However, we are printing out the config in the log, and that's normally what gets reported.

Reply:
Okay, we are indeed writing out the config to the derivatives folder:

https://github.com/poldracklab/fmriprep/blob/e3d3bc51dbf03215e3e4d2746d8aaacdd9afb84d/fmriprep/workflows/base.py#L79-L82

But now I'm realizing this is not a great place. This folder is generated and written at the subject level (regardless ``--bids-filters``, btw), and the problematic file producing the clobbering works at the run level. At the run level, one may want to process several participants, and thus how you decide the participant under which you will write this config?. I'm fine having written the config file underneath the processing of each subject under the subject's log folder (as we are currently doing), but a long term solution for this involves either writing to a temporary folder or tagging the name with the uuid. Then #2107 would need to adapt accordingly (I guess this could be done extracting the most recent config from the subjects involved in the processing).